src/quilla/plugins/blob_storage.py

```diff
@@ -0,0 +1,167 @@
+            help='The name of the container that will be used for storing Quilla images'
+        )
+
+        az_group.add_argument(
+            '--max-run-days',
+            dest='run_days',
+            type=int,
+            default=30,
+            help='The maximum number of days that reports should be allowed to exist. '
+            'Reports older than this amount of days will be deleted. Set to -1 to let '
+            'reports be kept indefinitely.'
+        )
+
+    def configure(
+        self,
+        connection_string: str,
+        container_name: str,
+        run_days: int,
+    ):
+        '''
+        Configure the container client and other necessary data, such as the max cleanup time.
+
+        If a container with that name does not exist, it will be created.
+
+        Args:
+            connection_string: The full connection string to the storage account
+            container_name: The name of the container that should be used to store all images
+            run_days: The maximum number of days a report should be allowed to have before being
+                cleaned up
+        '''
+        self._container_client = client = ContainerClient.from_connection_string(
+            connection_string,
+            container_name=container_name
+        )
+
+        self.max_run_days = run_days
+
+        if not client.exists():
+            client.create_container()
+
+    @property
+    def container_client(self) -> ContainerClient:
+        '''
+        An instance of the container client, casting it to ContainerClient.
+
+        This should be used exclusively from the abstract methods from BaseStorage
+
+        Returns:
+            The container client
+        '''
+
+        return cast(ContainerClient, self._container_client)
+
+    def quilla_configure(self, args: Namespace):
+        '''
+        Configures the plugin to run
+
+        Args:
+            args: A namespace generated by parsing the args from the CLI
+        '''
+        if args.connection_string is None:
+            return
+
+        self.configure(
+            args.connection_string,
+            args.container_name,
+            args.run_days,
+        )
+
+    @property
+    def is_enabled(self) -> bool:
+        return self._container_client is not None
+
+    def find_image_by_baseline(self, baseline_id: str) -> bytes:
+        blob = self.container_client.get_blob_client(f'baselines/{baseline_id}.png')
+
+        if not blob.exists():
+            return b''
+
+        downloader = blob.download_blob()
+
+        blob_data = downloader.readall()
+
+        return blob_data
+
+    def store_baseline_image(self, run_id: str, baseline_id: str, baseline: bytes) -> str:
+        blob = self.container_client.get_blob_client(f'baselines/{baseline_id}.png')
+        snapshot = self.container_client.get_blob_client(
+            f'baselines/snapshots/{run_id}/{baseline_id}.png'
+        )
+
+        blob.upload_blob(baseline)
```

Comment:
How to handle synchronization if multiple tests try to update baseline image at the same time?

Reply:
I'm not really sure that's a possibility as it stands- The most viable solution would be to say "run the baseline updates sequentially, not in parallel", since Quilla doesn't even support multiprocessing out of the box